dbms/src/Functions/FunctionsJSON.cpp

```diff
@@ -0,0 +1,320 @@
+#include <Functions/FunctionsJSON.h>
+
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypeTuple.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Functions/FunctionFactory.h>
+
+
+namespace DB
+{
+template <typename T>
+class JSONNullableImplBase
+{
+public:
+    static DataTypePtr getType() { return std::make_shared<DataTypeNullable>(std::make_shared<T>()); }
+
+    static Field getDefault() { return {}; }
+};
+
+class JSONHasImpl : public JSONNullableImplBase<DataTypeUInt8>
+{
+public:
+    static constexpr auto name{"jsonHas"};
+
+    static Field getValue(ParsedJson::iterator &) { return {1}; }
+};
+
+class JSONLengthImpl : public JSONNullableImplBase<DataTypeUInt64>
+{
+public:
+    static constexpr auto name{"jsonLength"};
```

Comment:
It's not easy to understand what `jsonLength()` is, and why `jsonLength('{"a":2,"b":35.4,"c":[1,2,{"e":"f\\n"}], "d":"qq"}')` is 8 and not 4.

Reply:
Could you please write some documentation and put it into https://github.com/yandex/ClickHouse/blob/master/docs/en/query_language/functions/json_functions.md ?

Reply:
Simdjson counts both keys and values when iterate through an object.

How about accessing keys only? So {"a":1,"b":2} will have `length = 2`, `[1] = "a"`, and `[2] = "b"`.

Reply:
@vitlibar Ok. I will push some updates tomorrow (GMT+8).

Reply:
Probably iterating this way would be slower. I suppose with documentation the current approach is ok.